protocol/validation/tx.go

```diff
@@ -22,6 +20,44 @@ var (
 	ErrFalseVMResult = errors.New("false VM result")
 )
 
+var (
+	// "suberrors" for ErrBadTx
+	errTxVersion              = errors.New("unknown transaction version")
```

Comment:
I'm not totally on board with defining these errors up here instead of where they're used. It would be clearly useful if each error were used multiple times, or if callers were testing the error value to handle them programmatically. But in this case it seems like an unnecessary level of indirection. What do you think?